src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java

```diff
@@ -0,0 +1,268 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.domains.Domain;
+import hudson.logging.LogRecorder;
+import hudson.logging.LogRecorderManager;
+import hudson.model.Slave;
+import hudson.util.Secret;
+import jenkins.plugins.git.GitSampleRepoRule;
+import jenkins.plugins.git.GitStep;
+import org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Formatter;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+import java.util.stream.Collectors;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.hamcrest.Matchers.*;
+
+public class GithubAppCredentialsTest extends AbstractGitHubWireMockTest {
+
+    private static Slave agent;
+    private static final String myAppCredentialsId = "myAppCredentialsId";
+    private static CredentialsStore store;
+    private static GitHubAppCredentials appCredentials;
+    private static LogRecorder logRecorder;
+
+    @Rule
+    public GitSampleRepoRule sampleRepo = new GitSampleRepoRule();
+
+    @BeforeClass
+    public static void setUpJenkins() throws Exception {
+        //Add credential (Must have valid private key for Jwt to work, but App doesn't have to actually exist)
+        store = CredentialsProvider.lookupStores(r.jenkins).iterator().next();
+        appCredentials = new GitHubAppCredentials(
+            CredentialsScope.GLOBAL, myAppCredentialsId, "sample", "54321", Secret.fromString(JwtHelperTest.PKCS8_PRIVATE_KEY));
+        appCredentials.setOwner("cloudbeers");
+        store.addCredentials(Domain.global(), appCredentials);
+
+        LogRecorderManager mgr = r.jenkins.getLog();
+        logRecorder = new LogRecorder(GitHubAppCredentials.class.getName());
+        mgr.logRecorders.put(GitHubAppCredentials.class.getName(), logRecorder);
+        LogRecorder.Target t = new LogRecorder.Target(GitHubAppCredentials.class.getName(), Level.FINER);
+        logRecorder.targets.add(t);
+        logRecorder.save();
+        t.enable();
```

Comment:
Did you just mean to use `LoggerRule`? Or are you doing it the hard way specifically to capture agent logs?

Reply:
Yes, I tried LoggerRule and it doesn't capture agent logs.  Otherwise, yes, I'd use LoggerRule.

Reply:
Got it. Probably something that could be improved in `LoggerRule`.